code/modules/integrated_electronics/subtypes/input.dm

```diff
@@ -1368,28 +1367,47 @@
 		activate_pin(1)
 
 /obj/item/integrated_circuit/input/tcomm_interceptor/on_data_written()
-	if(get_pin_data(IC_INPUT, 2))
-		GLOB.ic_jammers |= src
-		if(get_pin_data(IC_INPUT, 1))
-			power_draw_idle = 200
-		else
-			power_draw_idle = 100
+	if(get_pin_data(IC_INPUT, 1))
+		power_draw_idle = 100
 	else
-		GLOB.ic_jammers -= src
-		if(get_pin_data(IC_INPUT, 1))
-			power_draw_idle = 100
-		else
-			power_draw_idle = 0
+		power_draw_idle = 0
 
 /obj/item/integrated_circuit/input/tcomm_interceptor/power_fail()
 	set_pin_data(IC_INPUT, 1, 0)
-	set_pin_data(IC_INPUT, 2, 0)
 
 /obj/item/integrated_circuit/input/tcomm_interceptor/disconnect_all()
 	set_pin_data(IC_INPUT, 1, 0)
-	set_pin_data(IC_INPUT, 2, 0)
 	..()
 
+/obj/item/integrated_circuit/input/tcomm_interceptor/attackby(obj/O, mob/user)
+	if(istype(O, /obj/item/encryptionkey))
+		user.transferItemToLoc(O,src)
+		encryption_keys += O
+		recalculate_channels()
+		to_chat(user, "<span class='notice'>You slide \the [O] inside the circuit.</span>")
+	else
+		..()
+
+/obj/item/integrated_circuit/input/tcomm_interceptor/attack_self(mob/user)
+	if(encryption_keys.len)
+		for(var/i in encryption_keys)
+			var/obj/O = i
+			O.forceMove(drop_location())
+		encryption_keys.Cut()
+		to_chat(user, "<span class='notice'>You slide the encryption keys out of the circuit.</span>")
+		recalculate_channels()
+	else
+		to_chat(user, "<span class='notice'>There are no encryption keys to remove from the mechanism.</span>")
+
+/obj/item/integrated_circuit/input/tcomm_interceptor/proc/recalculate_channels()
+	whitelisted_freqs.Cut()
+	whitelisted_freqs.Add(FREQ_COMMON)
+	var/list/weakreffd_ekeys = list()
+	for(var/o in encryption_keys)
+		var/obj/item/encryptionkey/K = o
+		weakreffd_ekeys += WEAKREF(K)
```

Comment:
What is the purpose of having a local list of weakref to keys? Do you plan to use it?

Reply:
for ejecting the keys

Reply:
oh wait nevermind I have no fucking idea